### ISSUE TITLE
Using `realm-dart` alongside `RealmSwift`

### ISSUE THREAD

**Issue by putnokiabel** [realm/realm-dart#1195]
### What happened?

We're working on a production Flutter app (on iOS) that's currently using `RealmSwift`, and we'd would like to migrate to using [`realm`](https://pub.dev/packages/realm) instead.

However, we need to be able to use both of these side-by-side, since we have native code supporting the home screen widgets and lock screen widgets on iPhone as well as our Apple Watch companion app (which currently also use `RealmSwift`).

It does not seem possible to compile an iOS app using both, as `pod install` throws the following exception:
`The 'Pods-Runner' target has frameworks with conflicting names: realm.`

Is there a way to run both `RealmSwift` and `realm-dart` together?

### Repro steps

1. Create a fresh Flutter app supporting iOS (run `flutter create -t app --platforms ios test`).
2. Run `flutter add realm` from the app directory to add the `realm` package.
3. Add `pod 'RealmSwift', '10.36.0'` to `ios/Podfile` (or any other version of the pod for that matter).
4. Run `pod install` from the `ios` folder.
5. See the command fail with `The 'Pods-Runner' target has frameworks with conflicting names: realm.`.

The issue persists regardless of the versions of `realm` or `RealmSwift`.

### Version

1.0.2

### What Atlas Services are you using?

Local Database only

### What type of application is this?

Flutter Application

### Client OS and version

iOS 16

**Comment by nielsenko** [comment]
I must admit this was not a use-case we anticipated. I will investigate.

**Comment by nielsenko** [comment]
Apparently this should be possible, but right now the name clash prevents it. We will fix this in a coming release. Be aware that the different realm SDKs will still need to agree on the core file format version.

**Comment by nielsenko** [comment]
A few more caveats..

1. If you are using sync, then only one SDK can do the syncing. With flutter you can open the realm with `Configuration.disconnectedSync` which will allow you to open a synchronized realm without starting a sync session. I don't think swift allows this, but as long as one of the SDKs does then it should be good.
2. This relies on a feature of Mach-O where imported symbol are namespaced according to the shared object they come from - if they were using static linking then we’d be trying to cram two instances of Core in the same binary and they’d clobber each other.
3. The realm library will essentially be loaded twice in your process.

**Comment by putnokiabel** [comment]
@nielsenko Thank you for the quick reply, that sounds great! 
The caveats you've mentioned should not be a problem for our use-case, we're not using sync and were already counting on the library being loaded twice to make this work.

**Comment by putnokiabel** [comment]
@nielsenko Are there any plans on how (and when) to address the naming conflict with `RealmSwift`?

**Comment by nielsenko** [comment]
@putnokiabel This did not make it into 1.1.0, I'm sorry

**Comment by JYeop** [comment]
Same usecase here. Maybe the podspec name should be changed.

**Comment by JYeop** [comment]
Is there any plan to solve this issue?

**Comment by nirinchev** [comment]
Yes, we do intend to provide guidance on how to use Realm from multiple languages in the same app, but don't have timeframe for when that'll be available.

**Comment by rjeprasad** [comment]
@nielsenko This is a show-stopper for us. Please update ASAP. Any app with extension + realm will face this issue without a workaround. Just need a name change to remove the naming conflict. Rest should be fine for now.

**Comment by nielsenko** [comment]
@rjeprasad I'm sorry to hear that.

However, the original Dart team was laid of as part of a restructuring of Realm in September, and I no longer have any say in this, as I'm currently on garden leave. 

I see my old colleague @nirinchev, who is the lead on the .NET SDK, and who was helping out on the Dart team previously, has already commented on the road map.

**Comment by nirinchev** [comment]
We have this scheduled for later this year.

**Comment by ViolyCodevB** [comment]
Is there any progress on this issue?

**Comment by nielsenko** [comment]
@ViolyCodevB We are waiting for https://github.com/flutter/flutter/issues/63240 to be resolved, or native-assets to land. It looks like the latter will happen first.

As is, we would have to have to rename the `realm` package, which is not going to happen for this corner-case.

**Comment by nielsenko** [comment]
I created a [branch](https://github.com/realm/realm-dart/tree/kn/podspec-module-name) with the needed changes, but alas it won't work until https://github.com/flutter/flutter/issues/63240 is resolved.

**Comment by ramzitannous** [comment]
@nielsenko  @nirinchev  @rjeprasad  is there any solution for this, I'm facing this issue exactly ?

**Comment by sinhpn92** [comment]
Hello. Did this issue resolve? Or any suggest to solve? Thanks

**Comment by nielsenko** [comment]
@sinhpn92 Sorry, but no. I will make sure to update this issue if it happens. https://github.com/flutter/flutter/issues/63240 is still not resolved, [native assets](https://github.com/dart-lang/sdk/issues/50565) hasn't landed yet (still experimental), and we didn't rename the realm package in order to rename the pod either.

But put a 👍 on the. original issue, so we can monitor interest.
